File: leet-code/maxProfit.js
const { max } = require('lodash')

/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function (prices) {
  let min = Number.MAX_SAFE_INTEGER
  let maxP = 0
  for (let i = 0; i < prices.length; i++) {
    if (min > prices[i]) {
      min = prices[i]
    } else if (prices[i] - min > maxP) {
      maxP = prices[i] - min
    }
  }
  return maxP
}

var maxProfit = function (prices) {
  let maxP = 0
  for (let i = 0; i < prices.length; i++) {
    for (let j = i; j < prices.length; j++) {
      if (prices[j] - prices[i] > maxP) {
        maxP = prices[j] - prices[i]
      }
    }
  }
  return maxP
}

File: leet-code/maxProfit.go
package main

import "math"

func maxProfit(prices []int) int {
	min := math.MaxInt32
	maxP := 0
	for i := range prices {
		if min > prices[i] {
			min = prices[i]
		} else if maxP < prices[i]-min {
			maxP = prices[i] - min
		}
	}
	return maxP
}
